src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs

```diff
@@ -83,185 +103,196 @@ internal struct NameDeclarationInfo
                     return false;
                 }
 
-                var type = typeInferenceService.InferType(semanticModel, argument.SpanStart, objectAsDefault: false, cancellationToken: cancellationToken);
-                if (type != null)
-                {
-                    result = new NameDeclarationInfo(
+                var inferredTypes = typeInferenceService.GetTypeInferenceInfo(semanticModel, argument.SpanStart, cancellationToken: cancellationToken);
+                var types = inferredTypes.SelectAsArray(t => t.InferredType);
+                var parameters = inferredTypes.SelectAsArray(t => t.ParameterName).WhereAsArray(p => p != null);
```

Comment:
Could just be `Select` to avoid the extra `AsArray` work?

Reply:
Apparently WhereAsArray only takes ImmutableArray<T> as input. Switched to select/where/asarray